### PR TITLE
fix multichannel intensity_image extra_properties, addresses #5324

### LINKED ISSUE #5324: `regionprops` multichannel measurements don't work for `extra_properties`
## Description

`regionprops` measurments have been supporting multi-channel intensity images as of recently.
However, when combined with `extra_properties`-functions that also rely on intensity the `extra_properties` only appear to be dispatched for one channel (at least the result is not multi-channel).   

## Way to reproduce

Call `regionprops` with a multi-channel `intensity_image` and an `extra_properties`-function that relies on intensity.
Observe the result.
(will fill in more detail if needed, a bit in a rush atm).

## Version information
```python
3.8.5 (default, Sep  4 2020, 07:30:14) 
[GCC 7.3.0]
Linux-5.8.0-48-generic-x86_64-with-glibc2.10
scikit-image version: 0.18.1
numpy version: 1.18.1
```

### PR DESCRIPTION
**Pull request by VolkerH**
## Description
This would close #5324, see discussion there.

## Checklist
existing tests pass locally, but  still needs an additional test, working on that

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by VolkerH** [comment]
I believe the MacOS CI is broken, but things seem to pass on Linux, so this is ready for review. 
Maybe @jni can have a look.

File: skimage/measure/_regionprops.py [hunk 1]
```diff
@@ -255,7 +255,14 @@ def __getattr__(self, attr):
             # determine whether func requires intensity image
             if n_args == 2:
                 if self._intensity_image is not None:
-                    return func(self.image, self.intensity_image)
+                    if self._multichannel:
+                        multichannel_list = [func(self.image,
+                                                  self.intensity_image[..., i])
+                                             for i in range(
+                            self.intensity_image.shape[-1])]
+                        return np.stack(multichannel_list, axis=-1)
```
[the remark this answers is not in the document]
Comment (VolkerH) [reply skimage/measure/_regionprops.py:263]: I agree in principle as I usually like using `partial`. However, several existing occurences of multichannel intensity property measurements in this file use this `np.stack` - construct, so I chose to do the same as a matter of consistency.
In fact there is almost a bit much boilerplate, but I couldn't factor it out into something like `_multichannel_prop_calculation` as some of the built-in properties have additional required and channel-dependent arguments.

[the remark this answers is not in the document]
Comment (VolkerH) [reply skimage/measure/_regionprops.py:263]: Just to add, I'd be happy to integrate the proposed change though, I just wanted to add my reasoning (consistency with other parts of the code) for choosing this approach for consideration. Not sure whether the more elegant approach suggested trumps consistency, I'll leave that up to reviewers.

[the remark this answers is not in the document]
Comment (rfezzani) [reply skimage/measure/_regionprops.py:263]: Sorry that I didn't found this before the integration of the other parts :smile:. You can keep it as is, I understand the consistency argument. We can leave the larger refactoring to another PR if the change worth it (code readability and/or performance) :wink: .

**Review by rfezzani — APPROVED** [review]
Thank you @VolkerH

**Comment by mkcor** [comment]
> I believe the MacOS CI is broken, but things seem to pass on Linux, so this is ready for review.

@VolkerH just reran CI and everything passed this time :slightly_smiling_face:

**Review by grlee77 — APPROVED** [review]
Thanks @VolkerH, this looks good.

**Comment by VolkerH** [comment]
Thanks everyone for merging this quickly.

**Pull request merged**